module/reports/helper.js

```diff
@@ -0,0 +1,557 @@
+                    query["userId"] = userId
+                }
+
+                let chartObject = [];
+                let dateRange = await _getDateRangeofReport(reportType);
+                let endOf = dateRange.endOf;
+                let startFrom = dateRange.startFrom;
+
+                query['$or'] = [
+                    { "syncedAt": { $gte: new Date(startFrom), $lte: new Date(endOf) } },
+                    { "tasks": { $elemMatch: { syncedAt: { $gte: new Date(startFrom), $lte: new Date(endOf) } } } },
+                ]
+
+                if (programId) {
+                    query['programInformation._id'] = ObjectId(programId);
+                }
+
+                const projectDetails = await userProjectsHelper.projectDocument(
+                    query,
+                    ["title",
+                        "taskReport",
+                        "status",
+                        "tasks.status",
+                        "tasks.startDate",
+                        "tasks.endDate",
+                        "tasks.name",
+                        "categories",
+                        "startDate",
+                        "endDate"],
+                    []
+                );
+
+                if (!projectDetails.length > 0) {
+
+                    return resolve({
+                        message: CONSTANTS.apiResponses.REPORTS_DATA_NOT_FOUND,
+                        result: []
+                    })
+                }
+
+                if (getPdf == true) {
+
+                    let types = await this.types();
+                    let returnTypeInfo = types.result.filter(type => {
+                        if (type.value == reportType) {
+                            return type.label;
+                        }
+                    });
+
+                    let projectData = [];
+                    await Promise.all(projectDetails.map(project => {
+                        let projectInfo = {
+                            startDate: project.startDate,
+                            endDate: project.endDate,
+                            status: project.status,
+                            title: project.title,
+                            tasks: []
+                        }
+
+                        project.tasks.map(task => {
+                            projectInfo.tasks.push({
+                                startDate: task.startDate,
+                                endDate: task.endDate,
+                                status: task.status,
+                                title: task.name
+                            })
+                        })
+                        projectData.push(projectInfo);
+                    }));
+
+                    let data = {
+                        "reportType": returnTypeInfo[0].label,
+                        "projectDetails": projectData
+                    }
+                    let response = await dhitiService.viewFullReport(userToken, data);
```

Comment:
@rakeshSgr - Why we need the token to call dhiti service ? Can just userId work ?